### PR TITLE
Add endpoint for creating and editing rules

### PR DESCRIPTION
**Pull request by rhystmills**
## What does this change?

This PR adds endpoints for creating and editing rules in the rule manager.

The endpoints are:
- `/create` e.g. https://manager.typerighter.local.dev-gutools.co.uk/create
- `/update` e.g. https://manager.typerighter.local.dev-gutools.co.uk/update

The `create` endpoint expects a POST request with a JSON body that contains at minimum a ruleType.

An example request body might look like:
```json
{
    "ruleType": "regex",
    "pattern": "/example/",
    "description": "A perfectly fine word"
}
```

The `update` endpoint expects a POST request with a JSON body that contains at minimum an `id`.

An example request body might look like:
```json
{
    "pattern": "/embiggens/",
    "id": 53
}
```

The validation rules are pretty sparse currently - for `create` the only custom validation is checking that the ruleType is one of "regex", "languageToolXML" or "languageToolCore".

## How to test

Automated tests:

1. Run the application from the project root with `./script/start`
2. In another tab, run `sbt` from the project root
3. Use the `rule-manager` project with `project ruleManager` in the sbt shell
4. Run the tests with `test`
Do all the tests pass?

Manual testing:

You may want to do some testing of the endpoints using something like `Postman`. I found I needed to grab some cookies from the browser in order to be validated, specifically `PLAY_SESSION`, `guToolsAuth` and `gutoolsAuth-assym`.

I used `application/json` as the `Content-Type` header with the json listed at the top of this page as the request body.

### CODE REVIEW

File: apps/rule-manager/app/model/UpdateRuleForm.scala [hunk 1]
```diff
@@ -0,0 +1,38 @@
+package model
+
+import play.api.data.Form
+import play.api.data.Forms.{boolean, mapping, nonEmptyText, number, optional, text}
+
+object UpdateRuleForm {
+  val form = Form(
+    mapping(
+      "id" -> number(),
+      "ruleType" -> optional(text()),
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply apps/rule-manager/app/model/UpdateRuleForm.scala:10]: Ha, yes. I'll do that.

File: apps/rule-manager/conf/routes [hunk 2]
```diff
@@ -16,4 +16,9 @@ GET     /healthcheck                controllers.HomeController.healthcheck()
 +nocsrf
 POST    /refresh                    controllers.RulesController.refresh()
 +nocsrf
+POST    /create                     controllers.RulesController.create()
++nocsrf
+POST    /update                     controllers.RulesController.update()
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply apps/rule-manager/conf/routes]: Sounds great, now done.

File: apps/rule-manager/app/model/UpdateRuleForm.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (rhystmills) [reply apps/rule-manager/app/model/UpdateRuleForm.scala:9]: Not sure how to how to go about verifying an optional value but I'm looking into it.

[the remark this answers is not in the document]
Comment (rhystmills) [reply apps/rule-manager/app/model/UpdateRuleForm.scala:9]: Now done.

File: apps/rule-manager/app/db/DbRule.scala [hunk 3]
```diff
@@ -132,6 +135,82 @@ object DbRule extends SQLSyntaxSupport[DbRule] {
...
+      formRule: UpdateRuleForm
+  )(implicit session: DBSession = autoSession): Either[Result, DbRule] = {
+    val updatedRule = DbRule
+      .find(formRule.id)
+      .toRight(NotFound("Rule not found matching ID"))
+      .map(existingRule =>
+        new DbRule(
+          id = Some(formRule.id),
+          ruleType = formRule.ruleType.getOrElse(existingRule.ruleType),
+          pattern = formRule.pattern match {
```
[the remark this answers is not in the document]
Comment (rhystmills) [reply apps/rule-manager/app/db/DbRule.scala]: (Now done)

**Review by jonathonherbert — APPROVED** [review]
LGTM! Non-blocking – wonder if there's a way of drying up the constraint code.

**Pull request merged**